### PR TITLE
[sc-71014] Add setup and test action for Mulesoft

### PR DESCRIPTION
**Pull request by hoangmirs**
https://app.shortcut.com/jfc-global-templates/story/71014/setup-the-ci

## What happened 👀

- Add `setup` and `test` actions to support for Mulesoft
- Add some instructions to the readme

## Insight 📝

- As these actions could be reused, we should create a new repository to store them.
- `secrets` can not be used in composite actions, so we need to pass them as `inputs`

## Proof Of Work 📹

It can be run on https://github.com/nimblehq/jfc-global-store-proc-api/actions/runs/6311226462/job/17134810570

### CODE REVIEW

**Comment by andyduong1920** [comment]
Please add some information into the Readme as well @hoangmirs 👍

File: README.md [hunk 1]
```diff
@@ -1,12 +1,97 @@
...
+    <picture>
+      <source media="(prefers-color-scheme: dark)" srcset="https://assets.nimblehq.co/logo/dark/logo-dark-text-320.png">
+      <img alt="Nimble logo" src="https://assets.nimblehq.co/logo/light/logo-light-text-320.png">
+    </picture>
+  </a>
+</p>
 
-Project repository template to set up all public projects at [Nimble](https://nimblehq.co/)
+<h3 align="center">Mulesoft GitHub Actions</h3>
+<p>A collection of composite GitHub actions designed specifically for Mulesoft integration projects. These actions provide pre-built automation steps and workflows to streamline development, testing, and deployment processes.</p>
```
Comment (andyduong1920) [inline README.md]:
```suggestion
<p>A collection of composite GitHub actions designed specifically for Mulesoft projects. These actions provide pre-built automation steps and workflows to streamline development, testing, and deployment processes.</p>
```

**Review by andyduong1920 — APPROVED** [review]

**Comment by andyduong1920** [comment]
@hoangmirs should we have the `shared workflow` that combine 2 actions in this repo as well, so the consumer apps could either
- design the workflow by them self -> just use 2 actions from us
- or just reuse the existing workflow here :D

**Comment by andyduong1920** [comment]
> @hoangmirs should we have the `shared workflow` that combine 2 actions in this repo as well, so the consumer apps could either
> 
>     * design the workflow by them self -> just use 2 actions from us
> 
>     * or just reuse the existing workflow here :D

But I think this is quite simple, we no need to create the shared workflow here, just have these actions, so far most consumer will just have these 2 steps in the test workflow.

**Review by tednguyendev — APPROVED** [review]

**Comment by hoangmirs** [comment]
> @hoangmirs should we have the `shared workflow` that combine 2 actions in this repo as well, so the consumer apps could either
> 
> * design the workflow by them self -> just use 2 actions from us
> * or just reuse the existing workflow here :D

Yes, we can add that later. I haven't checked it we can add some attributes like `on:` and `concurrency:` to the shared workflow, so I did not add it. If we can share those attributes, I'll create a shared workflow and add to the readme

File: README.md (hunk 1 above)
Comment (hoangmirs) [reply README.md]: Changed in https://github.com/nimblehq/mulesoft-actions/pull/1/commits/18de2c0137ad2dddfcc650d4cc6353c169c92350

**Pull request merged**